src/xrpld/overlay/Slot.h

```diff
@@ -593,244 +330,181 @@ class Slots final
-        typename Peer::id_t,
-        std::tuple<PeerState, uint16_t, uint32_t, std::uint32_t>>
-    getPeers(PublicKey const& validator)
-    {
-        auto const& it = slots_.find(validator);
-        if (it != slots_.end())
-            return it->second.getPeers();
-        return {};
-    }
-
-    /** Get Slot's state */
-    std::optional<SlotState>
-    getState(PublicKey const& validator)
-    {
-        auto const& it = slots_.find(validator);
-        if (it != slots_.end())
-            return it->second.getState();
-        return {};
-    }
-
     /** Called when a peer is deleted. If the peer was selected to be the
      * source of messages from the validator then squelched peers have to be
      * unsquelched.
      * @param id Peer's id
      * @param erase If true then erase the peer
      */
     void
-    deletePeer(id_t id, bool erase);
+    deletePeer(Peer::id_t id, bool erase);
+
+    /** Called to register that a given validator was squelched for a given
+     * peer. It is expected that this method is called by SquelchHandler.
+     *
+     * @param validatorKey Validator public key
+     * @param peerID peer ID
+     */
+    void
+    squelchValidator(PublicKey const& validatorKey, Peer::id_t peerID);
+
+    void
+    onWrite(beast::PropertyStream::Map& stream) const;
 
 private:
     /** Add message/peer if have not seen this message
      * from the peer. A message is aged after IDLED seconds.
      * Return true if added */
     bool
-    addPeerMessage(uint256 const& key, id_t id);
+    addPeerMessage(uint256 const& key, Peer::id_t id);
+
+    /**
+     * Updates the last message sent from a validator.
+     * @param validator The validator public key
+     * @param peer The peer ID sending the message
+     * @return true if the validator was updated, false otherwise
+     */
+    std::optional<PublicKey>
+    updateConsideredValidator(PublicKey const& validator, Peer::id_t peer);
+
+    /** Remove all validators that have become invalid due to selection
+     * criteria
+     * @return zero or more validators that have been removed.
+     */
+    std::vector<PublicKey>
+    cleanConsideredValidators();
+
+    /** Checks whether a given validator is squelched.
+     * @param validatorKey Validator public key
+     * @return true if a given validator was squelched
+     */
+    bool
+    validatorSquelched(PublicKey const& validatorKey);
+
+    /** Checks whether a given peer was recently sent a squelch message for
+     * a given validator.
+     * @param validatorKey Validator public key
+     * @param peerID Peer id
+     * @return true if a given validator was squelched for a given peer
+     */
+    bool
+    peerSquelched(PublicKey const& validatorKey, Peer::id_t peerID);
 
     std::atomic_bool reduceRelayReady_{false};
 
-    hash_map<PublicKey, Slot<clock_type>> slots_;
-    SquelchHandler const& handler_;  // squelch/unsquelch handler
+    slots_map slots_;
+    slots_map untrusted_slots_;
```

Comment:
We use Hungarian notation for the data members.

Reply:
Done